asmcomp/amd64/emit.mlp

```diff
@@ -489,18 +521,26 @@ let emit_instr fallthrough i =
       add_used_symbol func;
       emit_call func;
       record_frame i.live false i.dbg ~label:label_after
-  | Lop(Itailcall_ind { label_after = _; }) ->
+  | Lop(Itailcall_ind { label_after; }) ->
       output_epilogue begin fun () ->
-        I.jmp (arg i 0)
+        I.jmp (arg i 0);
+        if Config.spacetime then begin
+          record_frame Reg.Set.empty false i.dbg ~label:label_after
+        end
       end
-  | Lop(Itailcall_imm { func; label_after = _; }) ->
-      if func = !function_name then
-        I.jmp (label !tailrec_entry_point)
-      else begin
-        output_epilogue begin fun () ->
-          add_used_symbol func;
-          emit_jump func
+  | Lop(Itailcall_imm { func; label_after; }) ->
+      begin
```

Comment:
I disagree I'm afraid; there's another conditional following afterwards ("if Config.spacetime ...").  I've already wasted quite a lot of time finding a bug which turned out to be due to missing begin/end in this area, so I think it's best to be conservative.